# **📊 Testes Paramétricos: O que são, quando usar e aplicações práticas**

## **✅ Introdução**

<p align="justify">
Os testes paramétricos são procedimentos estatísticos utilizados para realizar inferências sobre os dados quando certas premissas são atendidas — principalmente a de que os dados seguem uma distribuição normal e apresentam homogeneidade de variâncias.

<p align="justify">
São considerados mais poderosos que os testes não paramétricos, pois aproveitam ao máximo as informações contidas nos dados. Por isso, são amplamente usados nas áreas de saúde, ciências sociais, engenharia e negócios para tomar decisões baseadas em evidências.

**🎯 Objetivo deste material:**

* Apresentar os principais testes paramétricos;

* Explicar o que são, quando utilizar e suas hipóteses;

* Demonstrar a aplicação prática com exemplos reais utilizando Python.

**✅ Categorias de Testes Paramétricos Abordadas:**

1. Testes para comparação de médias

2. Testes para verificação de normalidade

3. Testes para variância e homogeneidade

## **✅ 1. Testes para Comparação de Médias**

Quando queremos comparar grupos para saber se há diferença significativa nas médias, recorremos a esses testes.

### **🔹 Teste t de Student**

O clássico e mais conhecido dos testes paramétricos, o teste t de Student permite comparar médias de uma ou mais amostras.

Existem três variações importantes:

#### **➡️ Teste t para uma amostra**

**O que é?**

É um teste estatístico que avalia se a média de uma amostra é estatisticamente diferente de um valor conhecido ou teórico. Ele ajuda a verificar se uma determinada característica medida em um grupo pode ser generalizada ou se ocorreu apenas por acaso.

**Quando usar?**

Quando quiser saber se a média de um grupo está diferente de um valor esperado.

**Exemplo:**

* Comparar o rendimento médio de um time com a meta estipulada.
* Verificar se o colesterol médio de uma população está dentro ou fora do recomendado pela OMS.

**Hipóteses:**

* H₀: a média da amostra é igual ao valor de referência.
* H₁: a média da amostra é diferente do valor de referência.

In [ ]:
# Exemplo Prático

## Carregando bibliotecas
import numpy as np
from scipy.stats import ttest_1samp

## Fixando semente e gerando dados
np.random.seed(0)
dados = np.random.normal(50, 10, 30)

## Aplicando o teste
stat, p = ttest_1samp(dados, popmean = 55)

## Exibindo o resultado
print(f'p-valor: {p:.4f}')

p-valor: 0.7781


**Conclusão:**

<p align="justify">
O Teste t para uma amostra resultou em um p-valor de 0.7781. Considerando um nível de significância de 0.05, como o p-valor obtido é maior que esse limite, não rejeitamos a hipótese nula. Assim, não há evidências suficientes para afirmar que a média da amostra difere do valor de referência de 55. Portanto, com base nos dados analisados, a média pode ser considerada estatisticamente igual ao valor proposto.

#### **➡️ Teste t para duas amostras independentes**

**O que é?**

É um teste estatístico que avalia se existe uma diferença significativa entre as médias de dois grupos que não possuem qualquer relação ou dependência entre si. Esse teste considera que as variâncias dos grupos podem ou não ser iguais, conforme verificado previamente.

**Quando usar?**

Quando se deseja comparar o desempenho, comportamento ou características de dois grupos distintos e independentes. Por exemplo:

* Comparar a produtividade de duas equipes que atuam em setores diferentes.
* Verificar se homens e mulheres apresentam médias diferentes de IMC (Índice de Massa Corporal).

**Hipóteses:**

* H₀: as médias são iguais.
* H₁: as médias são diferentes.

In [ ]:
# Exemplo Prático

## Carregando biblioteca
from scipy.stats import ttest_ind

## Gerando grupos de dados
grupo1 = np.random.normal(100, 15, 30)
grupo2 = np.random.normal(110, 15, 30)

## Aplicando o teste
stat, p = ttest_ind(grupo1, grupo2)

## Exibindo resultado
print(f'p-valor: {p:.4f}')

p-valor: 0.0013


**Conclusão:**

<p align="justify">
O Teste t para duas amostras independentes resultou em um p-valor de 0.0013. Considerando um nível de significância de 0.05, como o p-valor obtido é menor que esse limite, rejeitamos a hipótese nula. Assim, há evidências suficientes para afirmar que as médias dos dois grupos são significativamente diferentes. Portanto, com base nos dados analisados, podemos concluir que existe uma diferença estatística entre os grupos comparados.

#### **➡️ Teste t pareado (para amostras dependentes)**

**O que é?**

É um teste estatístico que compara as médias de duas medições relacionadas, verificando se houve diferença significativa entre elas. Muito usado em cenários de "antes e depois", onde as mesmas unidades (pessoas, objetos ou sistemas) são avaliadas em dois momentos ou condições.

**Quando usar?**

Quando se quer avaliar mudanças causadas por uma intervenção ou tratamento sobre o mesmo grupo ou elemento. Exemplo:

* Avaliar o efeito de um treinamento no desempenho dos funcionários comparando notas antes e depois.
* Testar a eficácia de um medicamento medindo a pressão arterial antes e após o uso.

**Hipóteses:**

* H₀: não há diferença entre as médias das duas medições (efeito nulo).
* H₁: há uma diferença significativa entre as médias (efeito presente).



In [ ]:
# Exemplo Prático

## Carregando biblioteca
from scipy.stats import ttest_rel

## Gerando dados
antes = np.random.normal(120, 10, 30)
depois = antes + np.random.normal(0, 5, 30)

## Aplicando teste
stat, p = ttest_rel(antes, depois)

## Imprimindo resultado
print(f'p-valor: {p:.4f}')

p-valor: 0.5739


**Conclusão:**

<p align="justify">
O Teste t pareado resultou em um p-valor de 0.5739. Considerando um nível de significância de 0.05, como o p-valor obtido é maior que esse limite, não rejeitamos a hipótese nula. Assim, não há evidências suficientes para afirmar que há uma diferença significativa entre as médias das medições antes e depois. Portanto, com base nos dados analisados, podemos concluir que não foi identificada uma diferença estatisticamente significativa entre as duas condições comparadas.

### **🔹 ANOVA (Análise de Variância)**

Quando se deseja comparar mais de dois grupos, o teste t não é mais suficiente. Para isso, usamos a ANOVA.

#### **➡️ ANOVA One-Way**

**O que é?**

É uma análise estatística que verifica se existe diferença significativa entre as médias de três ou mais grupos independentes. Ao invés de realizar múltiplos testes t (o que aumentaria o risco de erro), a ANOVA concentra tudo em uma única análise, indicando se ao menos um grupo se diferencia dos demais.

**Quando usar?**

Quando se deseja comparar mais de dois grupos que não possuem relação entre si. Exemplos:

* Comparar o desempenho de várias turmas escolares em uma mesma prova.
* Avaliar a eficácia de diferentes tratamentos médicos aplicados a grupos distintos de pacientes.

**Hipóteses:**

* H₀: todas as médias populacionais são iguais (não há diferença significativa).
* H₁: ao menos uma das médias é diferente das demais.

In [ ]:
# Exemplo Prático

## Carregando biblioteca
from scipy.stats import f_oneway

## Gerando grupos de dados
grupoA = np.random.normal(50, 5, 30)
grupoB = np.random.normal(55, 5, 30)
grupoC = np.random.normal(60, 5, 30)

## Aplicando o teste
stat, p = f_oneway(grupoA, grupoB, grupoC)

## Imprimindo resultado
print(f'p-valor: {p:.4f}')

p-valor: 0.0000


**Conclusão:**

<p align="justify">
O Teste ANOVA (Análise de Variância) resultou em um p-valor de 0.0000. Considerando um nível de significância de 0.05, como o p-valor obtido é menor que esse limite, rejeitamos a hipótese nula. Assim, há evidências suficientes para afirmar que pelo menos uma das médias dos grupos analisados é significativamente diferente. Portanto, com base nos dados, podemos concluir que existe uma diferença estatisticamente significativa entre as médias dos grupos comparados.

### **➡️ ANOVA Two-Way**

**O que é?**

É uma técnica estatística que investiga o impacto simultâneo de dois fatores (ou variáveis independentes) sobre uma variável dependente. Além de analisar os efeitos isolados de cada fator, também verifica se há interação entre eles — ou seja, se a combinação dos fatores produz um efeito diferente do que cada um causaria isoladamente.

**Quando usar?**

Quando se deseja compreender como dois aspectos influenciam um resultado, separadamente e em conjunto. Exemplo:

* Estudar o impacto de dieta e exercício físico na perda de peso.
* Avaliar o efeito de duas metodologias de ensino em diferentes turmas escolares.

**Hipóteses:**

* H₀: não há efeito significativo de nenhum dos fatores nem da interação entre eles.
* H₁: há efeito significativo de pelo menos um fator ou da interação.

In [ ]:
# Exemplo Prático

## Carregando bibliotecas
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols

## Construindo DataFrame
df = pd.DataFrame({
    'fator1': np.repeat(['A', 'B'], 30),
    'fator2': np.tile(['X', 'Y'], 30),
    'resposta': np.random.normal(50, 10, 60)
})

## Aplicando teste
modelo = ols('resposta ~ C(fator1) * C(fator2)', data = df).fit()
anova_twoway = sm.stats.anova_lm(modelo, typ = 2)

## Imprimindo resultado
print(anova_twoway)

                          sum_sq    df         F    PR(>F)
C(fator1)             377.657558   1.0  3.747750  0.057933
C(fator2)              63.178663   1.0  0.626964  0.431812
C(fator1):C(fator2)    21.400242   1.0  0.212369  0.646702
Residual             5643.071771  56.0       NaN       NaN


**Conclusão:**

O Teste ANOVA Two-way apresentou os seguintes resultados:

* Para o fator 1, o p-valor foi de 0.0579;

* Para o fator 2, o p-valor foi de 0.4318;

* Para a interação entre os fatores, o p-valor foi de 0.6467.

Considerando um nível de significância de 0.05:

* Para o fator 1: o p-valor é maior que 0.05, portanto não rejeitamos a hipótese nula. Não há evidências suficientes para afirmar que o fator 1 tem efeito significativo sobre a variável resposta.

* Para o fator 2: o p-valor também é maior que 0.05, logo não rejeitamos a hipótese nula. Não há evidências de que o fator 2 tenha efeito significativo.

* Para a interação: o p-valor novamente é maior que 0.05, indicando que não há evidências suficientes de interação significativa entre os fatores.

Assim, com base nos dados analisados, não foram encontradas diferenças estatisticamente significativas associadas aos fatores ou à interação entre eles.

### **✅ 2. Testes de Normalidade**

Antes de aplicar testes paramétricos, é essencial verificar se os dados seguem distribuição normal.

#### **🔹 Teste de Shapiro-Wilk**

**O que é?**

É um dos testes estatísticos mais utilizados para verificar se uma amostra de dados segue uma distribuição normal. Ele avalia a aderência dos dados à curva normal, sendo especialmente recomendado para amostras pequenas a moderadas.

**Quando usar?**

Sempre que for necessário verificar a suposição de normalidade antes de aplicar testes paramétricos, como o teste t ou a ANOVA. Exemplo:

* Checar se os tempos de reação seguem distribuição normal antes de aplicar um teste t.
* Validar a normalidade de notas escolares antes de realizar uma ANOVA.

**Hipóteses:**

* H₀: os dados seguem uma distribuição normal.
* H₁: os dados não seguem uma distribuição normal.

In [ ]:
# Exemplo Prático

## Carregando biblioteca
from scipy.stats import shapiro

## Aplicando teste
stat, p = shapiro(dados)

## Imprimindo resultado
print(f'p-valor: {p:.4f}')

p-valor: 0.5254


**Conclusão:**

<p align="justify">
O Teste de Shapiro-Wilk resultou em um p-valor de 0.5254. Considerando um nível de significância de 0.05, como o p-valor obtido é maior que esse limite, não rejeitamos a hipótese nula. Assim, não há evidências suficientes para afirmar que a amostra não segue uma distribuição normal, ou seja, os dados podem ser considerados normalmente distribuídos.

#### **🔹 Teste de Anderson-Darling**

**O que é?**

É um teste estatístico robusto para avaliar se uma amostra segue uma distribuição específica, geralmente a normal. Diferente de outros testes, ele dá mais peso às caudas da distribuição, tornando-se sensível a desvios nessas regiões.

**Quando usar?**

Quando se deseja uma avaliação mais rigorosa da normalidade, especialmente em análises onde desvios nas extremidades dos dados podem impactar os resultados.* Exemplo:

* Verificar se os resíduos de um modelo seguem distribuição normal.
* Checar normalidade em estudos financeiros, onde caudas extremas são importantes.

Hipóteses:

* H₀: os dados seguem a distribuição especificada (normal, na maioria dos casos).
* H₁: os dados não seguem a distribuição especificada.

In [ ]:
# Exemplo Prático

## Carregando biblioteca
from scipy.stats import anderson

## Aplicando teste
result = anderson(dados)

## Imprimindo resultado
print(f'Estatística: {result.statistic:.4f}')
print('Críticos:', result.critical_values)

Estatística: 0.2627
Críticos: [0.521 0.593 0.712 0.83  0.988]


**Conclusão:**

<p align="justify">
O Teste de Anderson-Darling resultou em uma estatística de 0.2627, enquanto os valores críticos para os níveis de significância comuns são [0.521, 0.593, 0.712, 0.83, 0.988]. Como a estatística obtida é menor que o valor crítico para o nível de 0.05 (0.712), não rejeitamos a hipótese nula. Dessa forma, não há evidências suficientes para afirmar que a amostra não segue uma distribuição normal, ou seja, os dados podem ser considerados normalmente distribuídos.

### **✅ 3. Testes para Variância e Homogeneidade**

Outra premissa importante para testes paramétricos é a homogeneidade das variâncias entre grupos.

#### **🔹 Teste F para igualdade de variâncias**

**O que é?**

É um teste estatístico que compara as variâncias de dois grupos, verificando se elas podem ser consideradas estatisticamente iguais. Ele é frequentemente utilizado como pré-requisito antes de testes que assumem homogeneidade de variância, como o teste t para duas amostras independentes.*

**Quando usar?**

Quando é necessário verificar se dois grupos apresentam a mesma variabilidade.* Exemplos:

* Antes de aplicar um teste t para duas amostras independentes.
* Ao comparar a consistência de desempenho entre dois times ou produtos.

**Hipóteses:**

* H₀: as variâncias dos dois grupos são iguais.
* H₁: as variâncias dos dois grupos são diferentes.

In [ ]:
# Exemplo Prático

## Gerando dados
var1 = np.var(grupo1, ddof=1)
var2 = np.var(grupo2, ddof=1)

## Calculando valor de F
F = var1 / var2

## Imprimindo resultado
print(f'Estatística F: {F:.4f}')

Estatística F: 0.8975


**Conclusão:**

<p align="justify">
O cálculo da estatística F para comparação das variâncias dos dois grupos resultou em 0.8975. Considerando que valores próximos de 1 indicam variâncias semelhantes entre os grupos, este resultado sugere que não há diferença significativa entre as variâncias das amostras analisadas.


#### **🔹 Teste de Levene**

**O que é?**

É um teste estatístico que avalia se diferentes grupos possuem variâncias iguais (homocedasticidade). É mais robusto do que o Teste F, pois é menos sensível à violação da suposição de normalidade.

**Quando usar?**

Antes de realizar testes que assumem homogeneidade de variância, especialmente quando há suspeita de que os dados não sejam normalmente distribuídos. Exemplos:

* Comparar a variabilidade do desempenho entre várias equipes.
* Avaliar se tratamentos diferentes têm a mesma dispersão de efeitos.

**Hipóteses:**

* H₀: as variâncias dos grupos são iguais.
* H₁: as variâncias dos grupos são diferentes.

In [ ]:
# Exemplo Prático

## Carregando biblioteca
from scipy.stats import levene

## Aplicando teste
_, p = levene(grupo1, grupo2)

## Imprimindo resultado
print(f'p-valor: {p:.4f}')

p-valor: 0.4982


**Conclusão:**

<p align="justify">
O teste de Levene para igualdade de variâncias resultou em um p-valor de 0.4982. Considerando um nível de significância de 0.05, como o p-valor obtido é maior que esse limite, não rejeitamos a hipótese nula. Assim, não há evidências suficientes para afirmar que as variâncias dos grupos são diferentes, indicando que as variâncias podem ser consideradas homogêneas.

# **✅ Considerações Finais**

✅ Os testes paramétricos são ferramentas fundamentais para quem quer realizar análises estatísticas com rigor.

✅ Eles oferecem maior poder estatístico, mas exigem que as premissas sejam respeitadas: normalidade, homocedasticidade e independência.

✅ Sempre que possível, complemente sua análise com gráficos (histogramas, boxplots) e os testes.

📌 Lembre-se: se as premissas não forem atendidas, opte por testes não paramétricos, como Mann-Whitney, Wilcoxon ou Kruskal-Wallis.

---
**Autor:** Rivaldo Correia Santos Júnior

**LinkedIn:** [https://www.linkedin.com/in/rivaldo-correia/]